spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/MicrosoftTeamsNotifier.java

```diff
@@ -0,0 +1,372 @@
+        if(imageUrl != null) {
+            section.setActivityImage(imageUrl.toString());
+        }
+        section.setActivitySubtitle(activitySubtitle);
+        section.getFacts().add(new Fact(STATUS_KEY, app.getStatusInfo().getStatus()));
+        section.getFacts().add(new Fact(SERVICE_URL_KEY, app.getServiceUrl()));
+        section.getFacts().add(new Fact(HEALTH_URL_KEY, app.getHealthUrl()));
+        section.getFacts().add(new Fact(MANAGEMENT_URL_KEY, app.getManagementUrl()));
+        section.getFacts().add(new Fact(SOURCE_KEY, app.getSource()));
+
+        message.getSections().add(section);
+
+        return message;
+    }
+
+    public MicrosoftTeamsNotifier() {
+        this.restTemplate = new RestTemplate();
+    }
+
+    public MicrosoftTeamsNotifier(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public void setWebhookUrl(URI webhookUrl) {
+        this.webhookUrl = webhookUrl;
+    }
+
+    public URI getImageUrl() {
+        return imageUrl;
+    }
+
+    public void setImageUrl(URI imageUrl) {
+        this.imageUrl = imageUrl;
+    }
+
+    public void setThemeColor(String themeColor) {
+        this.themeColor = themeColor;
+    }
+
+    public String getDeregisterActivitySubtitlePattern() {
+        return deregisterActivitySubtitlePattern;
+    }
+
+    public void setDeregisterActivitySubtitlePattern(String deregisterActivitySubtitlePattern) {
+        this.deregisterActivitySubtitlePattern = deregisterActivitySubtitlePattern;
+    }
+
+    public String getRegisterActivitySubtitlePattern() {
+        return registerActivitySubtitlePattern;
+    }
+
+    public void setRegisterActivitySubtitlePattern(String registerActivitySubtitlePattern) {
+        this.registerActivitySubtitlePattern = registerActivitySubtitlePattern;
+    }
+
+    public String getStatusActivitySubtitlePattern() {
+        return statusActivitySubtitlePattern;
+    }
+
+    public void setStatusActivitySubtitlePattern(String statusActivitySubtitlePattern) {
+        this.statusActivitySubtitlePattern = statusActivitySubtitlePattern;
+    }
+
+    public String getDeRegisteredTitle() {
+        return deRegisteredTitle;
+    }
+
+    public void setDeRegisteredTitle(String deRegisteredTitle) {
+        this.deRegisteredTitle = deRegisteredTitle;
+    }
+
+    public String getRegisteredTitle() {
+        return registeredTitle;
+    }
+
+    public void setRegisteredTitle(String registeredTitle) {
+        this.registeredTitle = registeredTitle;
+    }
+
+    public String getStatusChangedTitle() {
+        return statusChangedTitle;
+    }
+
+    public void setStatusChangedTitle(String statusChangedTitle) {
+        this.statusChangedTitle = statusChangedTitle;
+    }
+
+    public String getMessageSummary() {
+        return messageSummary;
+    }
+
+    public void setMessageSummary(String messageSummary) {
+        this.messageSummary = messageSummary;
+    }
+
+
+    @JsonSerialize
+    class Message {
```

Comment:
either don't make it an inner class or make it static

Reply:
Addressed in [95d0b934cb83ed50f29baefc70f583f901db4c87](https://github.com/codecentric/spring-boot-admin/pull/592/commits/95d0b934cb83ed50f29baefc70f583f901db4c87)